CHANGELOG.md

```diff
@@ -117,6 +117,7 @@
 - Display a clearer error message when `Index.names` is set to a non-like-like object.
 - Raise a warning whenever MultiIndex values are pulled in locally.
 - Improve warning message for `pd.read_snowflake` include the creation reason when temp table creation is triggered.
+- When calling `DataFrame.set_index`, or setting `DataFrame.index` or `Series.index`, with a new index that does not match the current length of the `Series`/`DataFrame` object, a `ValueError` is no longer raised. When the `Series`/`DataFrame` object is longer than the new index, the `Series`/`DataFrame`'s new index is filled with `NaN` values for the "extra" elements. When the `Series`/`DataFrame` object is shorter than the new index, the extra values in the new index are ignored—`Series` and `DataFrame` stay the same length `n`, and use only the first `n` values of the new index.
```

Comment:
@sfc-gh-azhan @sfc-gh-vbudati i am moving this from behavior change to improvement to avoid extra release process for behavior change.

Reply:
sounds good!

Reply:
Can we make this a bit less verbose? For example the last sentence seems redundant. Also, is this the behavior of native pandas? If so, let's start by something like:
```suggestion
- Conform with the behavior of native pandas when calling `DataFrame.set_index`, or setting `DataFrame.index` or `Series.index` with a new index that does not match the current length of the `Series`/`DataFrame` object. A `ValueError` is no longer raised. Instead, when the `Series`/`DataFrame` object is longer than the provided index, the `Series`/`DataFrame`'s new index is filled with `NaN` values for the "extra" elements. When the `Series`/`DataFrame` object is shorter than the provided index, the extra values are ignored in the new index.
```

Reply:
sure. the first sentence in the suggestion is not correct for the behavior since it actually diverge from the native pandas behavior, it is mainly for performance purpose to avoid eager evaluation. Updated it slightly differently